metadata/src/audio/item.rs

```diff
@@ -48,16 +48,17 @@ pub trait InnerAudioItem {
         restrictions: &Restrictions,
     ) -> AudioItemAvailability {
         let country = &user_data.country;
-        let user_catalogue = match user_data.attributes.get("catalogue") {
-            Some(catalogue) => catalogue,
-            None => "premium",
-        };
-
         for premium_restriction in restrictions.iter().filter(|restriction| {
             restriction
                 .catalogue_strs
                 .iter()
-                .any(|restricted_catalogue| restricted_catalogue == user_catalogue)
+                .any(|restricted_catalogue| {
+                    restricted_catalogue
+                        == user_data
+                            .attributes
+                            .get("catalogue")
```

Comment:
The reason I prefer the existing method, is that this needs to acquire a read lock in every iteration of the loop.

Reply:
I changed it because of a clippy lint. I don't remember what for? But regardless if you prefer it the other way I'll revert it and just tell clippy to ignore it.

Reply:
I don't know what did clippy complain about? Can we fix it in some other way, without acquiring a lock every time?

Reply:
> I don't know what did clippy complain about? Can we fix it in some other way, without acquiring a lock every time?

Like I said, I don't remember. I will have to revert it and then run it though clippy again.

Reply:
One thing at a time though. Let's deal with https://github.com/librespot-org/librespot/pull/965 and then I will deal with this PR.

Reply:
Well, I guess I'm not sure why I changed that. Turns out clippy is fine with it the way it was? In any event I reverted it.